lib/react_client/component_factory.dart

```diff
@@ -100,7 +98,7 @@ class ReactDartComponentFactoryProxy<TComponent extends Component> extends React
 
   /// Returns a JavaScript version of the specified [props], preprocessed for consumption by ReactJS and prepared for
   /// consumption by the `react` library internals.
-  static InteropProps generateExtendedJsProps(Map props, dynamic children, {Map defaultProps}) {
+  static InteropProps generateExtendedJsProps(Map props, dynamic children, {Map? defaultProps}) {
     if (children == null) {
       children = [];
```

Comment:
Should this be `const []` to avoid unnecessary redraws, or am I misunderstanding where/how this would be used by the JS pieces under the hood?

Reply:
Mm good question. I think using `const []` here would avoid unnecessary redraws, but only for legacy `Component` components that have a custom `shouldComponentUpdate` or custom `memo` equality function that doesn't do shallow equality checking of `children` items.

That feels like a pretty low amount of impact, so I'm reluctant to update this behavior in case consumers are incorrectly relying on components rerendering in those cases.